spec/views/posts/posts_show_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+RSpec.describe 'Post show', type: :feature do
+  describe 'Post' do
+    before(:each) do
+      @user1 = User.create(name: 'isaka', photo: 'Tom.png', bio: 'bio', posts_counter: 0, email: 'amy@gmail.com',
+                           password: 'password')
+      @user2 = User.create(name: 'Amy', bio: 'bio', photo: 'Tom.png', email: 'amy@gmail.com', password: 'password')
+      @user3 = User.create(name: 'Jerry', bio: 'bio', photo: 'Tom.png', email: 'jerry@gmail.com', password: 'password')
+
```

Comment:
- [ ] currently when I run these posts-show tests on this page all of them fail. please make sure that these tests are working to ensure your application runs as expected.

Reply:
hello, do you have any suggestion on this please? because am having a problem recognizing the problem, the tests are well written, the program is working as expected when I test it manually, but somehow tests don't pass can you please help me know what am doing wrong?

Reply:
I think looking through each of the tests that don't run and checking that you are taking the right steps for the test to run. for example, I noticed that in the `before(:each)` of your user tests. you are navigating to the root_page although you should be visiting the `user_sign_in_path`. because your root page doesn't have the login content.

with integration testing what you want to do is to match the steps you take when writing a test with what a user would be seeing on the page. 
so you simultaneously use your website as your write the tests to ensure what you are seeing is what you are writing.